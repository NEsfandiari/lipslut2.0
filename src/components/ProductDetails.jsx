import React, { Component } from 'react'
import styled from 'styled-components'
import ModalLayout from '../layouts/ModalLayout'
import ModalIngredients from './atoms/ModalIngredients'
import lightBlue from '../layouts/lightBlue'

const Container = styled.div`
  padding: 4rem 5rem 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  .ProductDetails-Content {
    display: flex;
    justify-content: space-between;
    width: 80%;
  }
  .ProductDetails-heading-5 {
    font-size: 1.4em;
    margin-bottom: 1rem;
  }

  .ProductDetails-paragraph {
    font-size: 0.9rem;
    margin-bottom: 2rem;
  }

  .ProductDetails-heading-6 {
    font-weight: bold;
    font-size: 0.9rem;
  }

  .ProductDetails-leftContent,
  .ProductDetails-rightContent {
    display: flex;
    flex-direction: column;
    flex-basis: 45%;
  }
  .ProductDetails-ingredientList-Link {
    color: blue;
    text-decoration: underline;
  }

  @media screen and (max-width: 420px) {
    padding: 1rem 0rem;
    .ProductDetails-Content {
      flex-direction: column;
      width: 100%;
    }
    .ProductDetails-leftContent,
    .ProductDetails-rightContent {
      margin: 2rem;
    }
  }
`

class ProductDetails extends Component {
  state = { hideList: true }

  toggleList = () => {
    this.setState(st => ({ hideList: !st.hideList }))
  }

  render() {
    let leftContent = []
    let rightContent = []
    const { claims } = this.props

    let keyIndex = claims.findIndex(element => {
      return element.content[0].value === 'Key Ingredients: '
    })

    for (let i = 0; i < keyIndex; i++) {
      leftContent.push(
        <div className={`ProductDetails-${claims[i].nodeType}`} key={i}>
          {claims[i].content[0].value}
        </div>
      )
    }

    for (let i = keyIndex; i < claims.length; i++) {
      //makes clickable button for ingredient list
      if (claims[i].content[0].value === 'Full ingredient list') {
        rightContent.push(
          <div
            onClick={this.toggleList}
            className={`ProductDetails-ingredientList-Link`}
            key={i}
          >
            {claims[i].content[0].value}
          </div>
        )
      } else {
        rightContent.push(
          <div className={`ProductDetails-${claims[i].nodeType}`} key={i}>
            {claims[i].content[0].value}
          </div>
        )
      }
    }
    return (
      <Container>
        <div className="ProductDetails-Content">
          <div className="ProductDetails-leftContent">{leftContent}</div>
          <div className="ProductDetails-rightContent">{rightContent}</div>
        </div>

        {this.state.hideList || (
          //ModalLayout component is rendered with a portal
          //everything inside ModalLayout is rendered at the top level (index.js)
          <ModalLayout>
            <ModalIngredients
              ingredients={this.props.ingredients}
              toggleList={this.toggleList}
            />
          </ModalLayout>
        )}
      </Container>
    )
  }
}

export default lightBlue(ProductDetails)
