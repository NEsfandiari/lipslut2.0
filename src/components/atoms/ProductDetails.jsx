import React, { Component } from 'react'
import { FaTimes } from 'react-icons/fa'
import Modal from '../Modal'
import styled from 'styled-components'
import lightBlue from '../../layouts/lightBlue'

const Container = styled.div`
  padding: 4rem 5rem 0rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  .ProductDetails-Content {
    display: flex;
    justify-content: space-between;
    width: 80%;
  }
  .ProductDetails-heading-5 {
    font-size: 1.4em;
    margin-bottom: 1rem;
  }

  .ProductDetails-paragraph {
    font-size: 0.9rem;
    margin-bottom: 2rem;
  }

  .ProductDetails-heading-6 {
    font-weight: bold;
    font-size: 0.9rem;
  }

  .ProductDetails-leftContent,
  .ProductDetails-rightContent {
    display: flex;
    flex-direction: column;
    flex-basis: 45%;
  }
  .ProductDetails-ingredientList-Link {
    color: blue;
    text-decoration: underline;
  }

  @media screen and (max-width: 420px) {
    padding: 1rem 0rem;
    .ProductDetails-Content {
      flex-direction: column;
      width: 100%;
    }
    .ProductDetails-leftContent,
    .ProductDetails-rightContent {
      margin: 2rem;
    }
  }
`

//inline style for modal background (used inline style so we did not need to add more styles to top level index.css)
//styled components do not apply due to use of portal
const style = {
  display: 'flex',
  alignItems: 'center',
  position: 'fixed',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: '100vw',
  height: '100vh',
  backgroundColor: 'rgba(255, 255, 255, 0.9)',
  zIndex: 100000,
}

const ingredientsStyle = {
  border: '2px solid black',
  padding: '2em',
  backgroundColor: 'white',
  width: '70%',
  height: 'fit-content',
  margin: '0 auto',
}

class ProductDetails extends Component {
  constructor(props) {
    super(props)
    this.state = { hideList: true }
    this.toggleList = this.toggleList.bind(this)
  }

  toggleList() {
    this.setState(st => ({ hideList: !st.hideList }))
  }

  render() {
    let leftContent = []
    let rightContent = []
    let { claims } = this.props.product

    let keyIndex = claims.findIndex(element => {
      return element.content[0].value === 'Key Ingredients: '
    })

    for (let i = 0; i < keyIndex; i++) {
      leftContent.push(
        <div className={`ProductDetails-${claims[i].nodeType}`} key={i}>
          {claims[i].content[0].value}
        </div>
      )
    }

    for (let i = keyIndex; i < claims.length; i++) {
      //makes clickable button for ingredient list
      if (claims[i].content[0].value === 'Full ingredient list') {
        rightContent.push(
          <div
            onClick={this.toggleList}
            className={`ProductDetails-ingredientList-Link`}
            key={i}
          >
            {claims[i].content[0].value}
          </div>
        )
      } else {
        rightContent.push(
          <div className={`ProductDetails-${claims[i].nodeType}`} key={i}>
            {claims[i].content[0].value}
          </div>
        )
      }
    }
    return (
      <Container>
        <div className="ProductDetails-Content">
          <div className="ProductDetails-leftContent">{leftContent}</div>
          <div className="ProductDetails-rightContent">{rightContent}</div>
        </div>

        {this.state.hideList ? (
          ''
        ) : (
          //Modal component is rendered with a portal
          //everything inside Modal is rendered at the top level (index.js)
          <Modal>
            <div style={style}>
              <div id="ingredientsBox" style={ingredientsStyle}>
                <div
                  style={{
                    display: 'flex',
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                  }}
                >
                  <div>
                    <h3>Ingredients</h3>
                  </div>
                  <div>
                    <button
                      style={{
                        backgroundColor: 'none',
                        border: 'none',
                        outline: 0,
                      }}
                      onClick={this.toggleList}
                    >
                      <FaTimes />
                    </button>
                  </div>
                </div>
                <p style={{ fontSize: '12px', lineHeight: '20px' }}>
                  {this.props.product.ingredients}
                </p>
              </div>
            </div>
          </Modal>
        )}
      </Container>
    )
  }
}

export default lightBlue(ProductDetails)
