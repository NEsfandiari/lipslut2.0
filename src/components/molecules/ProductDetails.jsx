import React, { Component } from './react'

export default class ProductDetails extends Component {
  render() {
    return
  }
}

/*{
  contentfulProductPage(title: {eq: "BATCHâ€”001: \"04\""}) {
    claims {
      nodeType
      content {
        nodeType
        content {
          value
          marks{
            type
          }
        }
      }
    }
    ingredients {
      nodeType
      content {
        content {
          value
        }
      }
    }
  }
}
*/
