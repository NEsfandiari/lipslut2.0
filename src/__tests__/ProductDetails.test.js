import React from 'react'
import renderer from 'react-test-renderer'
import ProductDetails from '../components/atoms/ProductDetails'

describe('ProductDetails', () => {
  it('renders correctly', () => {
    const data = {
      claims: [
        { nodeType: 'heading-5', content: [{ value: 'What is it?' }] },
        {
          nodeType: 'paragraph',
          content: [
            { value: 'A long wearing liquid lipstick with a matte finish.' },
          ],
        },
        { nodeType: 'heading-5', content: [{ value: 'How to use:' }] },
        {
          nodeType: 'paragraph',
          content: [
            {
              value:
                'Smooth a thin layer onto lips user our angled applicator for a bold pop of all day color. ',
            },
          ],
        },
        { nodeType: 'heading-5', content: [{ value: 'Claims:' }] },
        {
          nodeType: 'paragraph',
          content: [{ value: 'Cruelty Free, Paraben Free, Vegan. ' }],
        },
        { nodeType: 'heading-5', content: [{ value: 'Key Ingredients: ' }] },
        {
          nodeType: 'heading-6',
          content: [{ value: 'Color-Lock Technology' }],
        },
        {
          nodeType: 'paragraph',
          content: [
            {
              value:
                'One-of-a-kind, lightweight and creamy formula locks pigment onto lips for a high payoff look that wonâ€™t crease, fade, or smudge all day. ',
            },
          ],
        },
        {
          nodeType: 'heading-6',
          content: [{ value: 'Coconut + Jojoba Oil' }],
        },
        {
          nodeType: 'paragraph',
          content: [
            {
              value:
                'Makes for a silky application that spreads smooth and even. Sticky goop can f*ck off!',
            },
          ],
        },
        { nodeType: 'heading-6', content: [{ value: 'Vitamin E' }] },
        {
          nodeType: 'paragraph',
          content: [
            {
              value:
                'Leaves lips extra moisturized and plush. Helps provide a velvety seamless finish without flakiness or streakiness. ',
            },
          ],
        },
        {
          nodeType: 'paragraph',
          content: [{ value: 'Full ingredient list' }, { value: ' ' }],
        },
        { nodeType: 'paragraph', content: [{ value: '' }] },
        { nodeType: 'paragraph', content: [{ value: '' }] },
      ],
      ingredients:
        'Isododecane, Dimethicone, Microcrystalline Wax, Bentonite, Methyl Methacrylate Crosspolymer, Polymethylsilsesquioxane, Propylene Carbonate, Phenoxyethanol, Buxus Chinensis Oil, Coconut Oil, Tridecyl trimellitate, Titanium Dioxide, Red 28 Lake(CI 45410), Yellow 5 Lake.',
    }

    const tree = renderer.create(<ProductDetails product={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
