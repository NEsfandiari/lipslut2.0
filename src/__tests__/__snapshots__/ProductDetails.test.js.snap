// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails renders correctly 1`] = `
<div
  className="sc-bdVaJa iylCwT"
>
  <div
    className="sc-bwzfXH exGvLt"
  >
    <div
      className="ProductDetails-Content"
    >
      <div
        className="ProductDetails-leftContent"
      >
        <div
          className="ProductDetails-heading-5"
        >
          What is it?
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          A long wearing liquid lipstick with a matte finish.
        </div>
        <div
          className="ProductDetails-heading-5"
        >
          How to use:
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          Smooth a thin layer onto lips user our angled applicator for a bold pop of all day color. 
        </div>
        <div
          className="ProductDetails-heading-5"
        >
          Claims:
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          Cruelty Free, Paraben Free, Vegan. 
        </div>
      </div>
      <div
        className="ProductDetails-rightContent"
      >
        <div
          className="ProductDetails-heading-5"
        >
          Key Ingredients: 
        </div>
        <div
          className="ProductDetails-heading-6"
        >
          Color-Lock Technology
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          One-of-a-kind, lightweight and creamy formula locks pigment onto lips for a high payoff look that wonâ€™t crease, fade, or smudge all day. 
        </div>
        <div
          className="ProductDetails-heading-6"
        >
          Coconut + Jojoba Oil
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          Makes for a silky application that spreads smooth and even. Sticky goop can f*ck off!
        </div>
        <div
          className="ProductDetails-heading-6"
        >
          Vitamin E
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          Leaves lips extra moisturized and plush. Helps provide a velvety seamless finish without flakiness or streakiness. 
        </div>
        <div
          className="ProductDetails-ingredientList-Link"
          onClick={[Function]}
        >
          Full ingredient list
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          
        </div>
        <div
          className="ProductDetails-paragraph"
        >
          
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
